core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,1002 @@
+                # Validates various parts of FractionInput.
+                if value.interaction.id == 'FractionInput':
+                    inputs_with_whole_nums = [
+                        'HasDenominatorEqualTo',
+                        'HasNumeratorEqualTo',
+                        'HasIntegerPartEqualTo',
+                        'HasNoFractionalPart'
+                    ]
+                    allow_non_zero_integ_part = (
+                        value.interaction.customization_args[
+                            'allowNonzeroIntegerPart'].value)
+                    allow_imp_frac = (
+                        value.interaction.customization_args[
+                            'allowImproperFraction'].value)
+                    allow_req_simple_form = (
+                        value.interaction.customization_args[
+                            'requireSimplestForm'].value)
+
+                    for rule_spec in answer_group.rule_specs:
+                        rule_spec_index = str(answer_group.rule_specs.index(
+                            rule_spec))
+
+                        if rule_spec.rule_type not in inputs_with_whole_nums:
+                            num = rule_spec.inputs['f']['numerator']
+                            den = rule_spec.inputs['f']['denominator']
+                            whole = rule_spec.inputs['f']['wholeNumber']
+
+                            # Validates if the denominator value is zero.
+                            if den == 0:
+                                fraction_interaction_invalid_values.append(
+                                    'The rule {rule} of answer '.format(
+                                        rule=rule_spec_index) +
+                                    'group {answer_group} has '.format(
+                                        answer_group=ans_group_index) +
+                                    'denominator equals to zero.')
+
+                            # Validates if the value not in simplest form.
+                            if allow_req_simple_form is True:
+                                if whole == 0:
+                                    d = math.gcd(num, den)
+                                    val_num = num // d
+                                    val_den = den // d
+                                if val_num != num and val_den != den:
+                                    fraction_interaction_invalid_values.append(
+                                        'The rule {rule} of answer '.format(
+                                            rule=rule_spec_index) +
+                                        'group {answer_group} do '.format(
+                                            answer_group=ans_group_index) +
+                                        'not have value in simple form')
+
+                            # Validates if the value not in proper frac form.
+                            if allow_imp_frac is False:
+                                if den <= num:
```

Comment:
No, actually this is a separate check from the `simplest form` because it totally depends on whether the user has clicked `allow improper fractions` or not also it's mentioned separately in the issue desc as well - 
`All rules should have solutions in proper form if the allow improper fraction setting is turned off`
Thanks

Reply:
Can you maybe explain what are these different settings for fractions, so that we can just be 100% sure we are validating correct thing?

Reply:
Yeah sure,
So basically we do have 3 checkboxes present in the `FractionInput` interaction which user can click and those are - 
- Allow non zero integer parts
- Allow improper fractions
- Allow values in simple form

and all the checks are accordingly implemented, for eg if the user clicks `allow non zero integers` then the values greater than zero are valid otherwise not.
Thanks

Reply:
Hey, I'm not sure what all of these mean exactly, except maybe for the non zero integer parts. What is exactly simple form and what is improper fraction.

Reply:
Oh so basically the `Allow values in simple form` mean when we have a fraction value in the reduced form for example `3/4` is the reduced form but `2/4` is not because it can further be reduced to `1/2`.

And an improper fraction is a type of fraction where the numerator is greater than or equal to the denominator.
Thanks